coding/src/templatize.py

```diff
@@ -191,11 +221,16 @@ def templatize_instance(self, input, category):
     def generate_exemplars(self, n_exemplars, seed_exemplars=0):
         '''
         Select n randomly generated exemplars.
+        Arguments:
+            n_exemplars (int): number of exemplars to generate
+            seed_exemplars (int): random seed for drawing exemplars
         '''
-        # TODO - add functionality for sampling "ambiguous" or "prototypical"
         # offset seed so that exemplars are different than instances
         seed_offset = 42
-        exemplars = self.dataset.sample(n=n_exemplars, random_state=seed_exemplars+seed_offset)
+        if self.exemplar_method != 'random':
+            exemplars = self.candidate_exemplars.head(n_exemplars)
```

Comment:
Where are we filtering these exemplars by type? I believe it's happening but I'm confused

Reply:
In `set_exemplar` method